demo/admin/src/news/handmade/NewsForm.tsx

```diff
@@ -0,0 +1,337 @@
+    Toolbar,
+    ToolbarActions,
+    ToolbarFillSpace,
+    ToolbarItem,
+    ToolbarTitleItem,
+    useFormApiRef,
+    useStackApi,
+    useStackSwitchApi,
+} from "@comet/admin";
+import { FinalFormDatePicker } from "@comet/admin-date-time";
+import { ArrowLeft } from "@comet/admin-icons";
+import { AdminComponentRoot, AdminTabLabel, BlockState, createFinalFormBlock } from "@comet/blocks-admin";
+import {
+    BlockPreviewWithTabs,
+    DamImageBlock,
+    EditPageLayout,
+    queryUpdatedAt,
+    resolveHasSaveConflict,
+    useBlockPreview,
+    useCmsBlockContext,
+    useFormSaveConflict,
+    useSiteConfig,
+} from "@comet/cms-admin";
+import { IconButton, MenuItem } from "@mui/material";
+import { useContentScope } from "@src/common/ContentScopeProvider";
+import { FormApi } from "final-form";
+import { filter } from "graphql-anywhere";
+import isEqual from "lodash.isequal";
+import React from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { useRouteMatch } from "react-router";
+
+import { NewsContentBlock } from "../blocks/NewsContentBlock";
+import { createNewsMutation, newsFormFragment, newsFormQuery, updateNewsMutation } from "./NewsForm.gql";
+import {
+    GQLCreateNewsMutation,
+    GQLCreateNewsMutationVariables,
+    GQLNewsFormFragment,
+    GQLNewsFormQuery,
+    GQLNewsFormQueryVariables,
+    GQLUpdateNewsMutation,
+    GQLUpdateNewsMutationVariables,
+} from "./NewsForm.gql.generated";
+
+const rootBlocks = {
+    image: DamImageBlock,
+    content: NewsContentBlock,
+};
+
+type FormValues = GQLNewsFormFragment & {
+    image: BlockState<typeof rootBlocks.image>;
+    content: BlockState<typeof rootBlocks.content>;
+};
+
+interface FormProps {
+    id?: string;
+}
+
+export function NewsForm({ id }: FormProps): React.ReactElement {
+    const stackApi = useStackApi();
+    const client = useApolloClient();
+    const mode = id ? "edit" : "add";
+    const formApiRef = useFormApiRef<FormValues>();
+    const stackSwitchApi = useStackSwitchApi();
+    const { scope } = useContentScope();
+    const siteConfig = useSiteConfig({ scope });
+    const intl = useIntl();
+    const previewApi = useBlockPreview();
+    const match = useRouteMatch();
+    const blockContext = useCmsBlockContext();
+
+    const previewContext = {
+        ...blockContext,
+        parentUrl: match.url,
+        showVisibleOnly: previewApi.showOnlyVisible,
+    };
+
+    const { data, error, loading, refetch } = useQuery<GQLNewsFormQuery, GQLNewsFormQueryVariables>(
+        newsFormQuery,
+        id ? { variables: { id } } : { skip: true },
+    );
+
+    const initialValues = React.useMemo<Partial<FormValues>>(
+        () =>
+            data?.news
+                ? {
+                      ...filter<GQLNewsFormFragment>(newsFormFragment, data.news),
+                      image: rootBlocks.image.input2State(data.news.image),
+                      content: rootBlocks.content.input2State(data.news.content),
```

Comment:
`date` must be converted to a `Date`, otherwise the DatePicker can't handle it. We could also think about adjusting the DatePicker so it can handle a date string. 

https://github.com/vivid-planet/comet/assets/13380047/d7456e80-38ca-40b8-9dba-c620d2723337


What do you think?